packages/openid4vc/src/shared/utils.ts

```diff
@@ -107,6 +154,37 @@ export function getCreateJwtCallback(
       return jws
     }
 
+    if (jwtIssuer.method === 'custom') {
+      const { options } = jwtIssuer
+      if (!options) throw new CredoError(`Custom jwtIssuer must have options defined.`)
+      if (!options.clientId) throw new CredoError(`Custom jwtIssuer must have clientId defined.`)
+      if (typeof options.clientId !== 'string') throw new CredoError(`Custom jwtIssuer's clientId must be a string.`)
+
+      const { clientId } = options
+
+      const entityConfiguration = await fetchEntityConfiguration({
+        entityId: clientId as string,
+        verifyJwtCallback: async ({ data, signature, jwk }) => {
+          const jws = `${TypedArrayEncoder.toUtf8String(data)}.${TypedArrayEncoder.toBase64URL(signature)}`
+          const res = await jwsService.verifyJws(agentContext, { jws, jwkResolver: () => getJwkFromJson(jwk) })
+          return res.isValid
+        },
+      })
+
+      // TODO: Not 100% sure what key to pick here I think the one that matches the kid in the jwt header of the entity configuration or we should pass a alg and pick a jwk based on that?
+      const jwk = getJwkFromJson(entityConfiguration.jwks.keys[0])
+
+      // TODO: This gives a weird error when the private key is not available in the wallet
+      const jws = await jwsService.createJwsCompact(agentContext, {
+        protectedHeaderOptions: { ...jwt.header, jwk, alg: jwk.supportedSignatureAlgorithms[0] },
```

Comment:
the alg is also in the entity configuration jwks.key entry? Should we extract it from there first?

Reply:
Also, where do we use this callback for ? The keys in the entity configuration should ONLY be used for federation protocols. Once we do OID4VCI / OID4VP we should use different keys i think? Although im not sure which keys. 🤔 

> These Federation Entity Keys SHOULD NOT be used in other protocols. (Keys to be used in other protocols, such as OpenID Connect, are conveyed in the metadata elements for the protocol's Entity Type Identifiers, such as the metadata under the openid_provider and openid_relying_party Entity Type Identifiers.)

Reply:
I've updated it now the jwks of the openid_relying_party is being used.